### PR TITLE
Connect on Initialize Fix

### PR DESCRIPTION
**Pull request by rob1997**
There were issues in async execution order for web3auth in async plus some unhandled errors/exceptions

Bug was that connect would get stuck while trying to restore web3auth session

### CODE REVIEW

**Review by sneakzttv — APPROVED** [review]
fun way to pause for a frame, please add a description to the issue though., looks messy without it.

image

File: Packages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthProvider.cs [hunk 1]
```diff
@@ -93,7 +93,15 @@ private void Cancel()
...
     {
-        _connectTcs.SetResult(response);
+        if (string.IsNullOrEmpty(response.error))
+        {
+            _connectTcs.SetResult(response);
+        }
+
+        else
+        {
+            _connectTcs.SetException(new Web3Exception(response.error));
```
Comment (kantagara) [inline Packages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthProvider.cs:103]: Do we want to throw it in the users face? I am thinking we should silently fail but don't have strong opinion on it.🤷‍♂️

Comment (rob1997) [reply Packages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthProvider.cs:103]: when trying to restore session, it'll just log an error

**Review by kantagara — APPROVED** [review]

**Comment by rob1997** [comment]
> Fun unity specific way to pause for a frame, please add a description to the issue though. It looks messy without it.
> 
> image

good call, I've updated the description of the PR

**Review by juans-chainsafe — APPROVED** [review]

**Comment by rob1997** [comment]
Emm this was merged as part of #1154

**Pull request closed without merging**